### DISCUSSION TITLE
Minor update for a docker CKAN deployment

### DISCUSSION THREAD

**Discussion by doylejg** [ckan/ckan#7375] [Q&A]
I've taken over sysadmin responsibilities for a 2.8.5 ckan installation which I would like to upgrade to 2.9 (and eventually 2.10). The docs only describe using `.deb` files for performing a minor upgrade -- however ckan is in a docker container and furthermore the os is non debian (cent-os). 

Q: To upgrade, should I just pull the 2.9 version of the docker config from github and launch with the same config files? Are there migration issues I should be aware of? 

Thanks for your help.

**Comment by kowh-ai** [comment]
Definitely a good idea to use a more recent version of CKAN!...Is it _just_ CKAN in a container? What about the database? Any extensions included in the CKAN container? How many VM's (or physical machines) is everything spread across? Is it possible to run up new containers (with different container names and ports) to test?

**Comment by doylejg** [comment]
Database is in a separate container, I don't think there are extensions, one VM.

Yes it is possible to spin up a new container, I guess I was partially wondering if they containers would update themselves or if each version had a separate container. And in the case of the later, are there any migration notes I should know about between the two versions?

**Reply by kowh-ai** [reply]
With regard to just the CKAN container, you wouldn't be able to have 2 containers with the same container name **_or_** 2 containers with different names, but listening on the same port. If you want to use the same port and name you would remove the CKAN container (and local image) and build a new image (new version) and run this new container. You would need to use the Dockerfile for the new version of CKAN to build and to run. It could get a little tricky.

There is a section on migrating data in the **Install CKAN with Docker** doco: https://docs.ckan.org/en/2.9/maintaining/installing/install-from-docker-compose.html#migrate-data - but I have not tried this

**Reply by doylejg** [reply]
Thanks, I'm mostly just looking for guidance on upgrading from 2.8 to 2.9 assuming the docker install. I think what I'm getting is that I can just check out the newer image, build, and should be fine.

So far as I can see, the data migration was for a non-docker deployment to a docker deployment, which isn't the case here. Is that correct?

**Reply by kowh-ai** [reply]
With regard to the **resource files** in the doco, it shouldn't matter if the source is a non-docker or docker...at the end of the day they are just directories and files

**Reply by doylejg** [reply]
Yes they are. Thanks i guess you answered my question.
